community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/SchemaRead.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.kernel.api;
+
+/**
+ * Surface for getting schema information, such as fetching specific indexes or constraints.
+ */
+public interface SchemaRead
+{
+    IndexReference index( int label, int... properties );
```

Comment:
What if there is no such index? What about constraints?

Reply:
Eh, what about constraints? Cypher doesn't really care about them apart from create/drop, so let's add any constrainty methods on need?